### ISSUE TITLE
Detect assignment operators that do nothing (e.g. `+= 0`)

### ISSUE THREAD

**Issue by kpreid** [rust-lang/rust-clippy#16977]
### What it does

Currently, `identity_op` can detect expressions that contain useless operators:

```rust
warning: this operation has no effect
 --> src/lib.rs:3:15
  |
3 |     counter = counter + 0;
  |               ^^^^^^^^^^^ help: consider reducing it to: `counter`
```

However, nothing detects a useless assignment version of the same operation: `counter += 0` produces no lint at all. Detecting this seems to me to be a straightforward extension of what `identity_op` is already doing — though perhaps it should be a separate lint, since `identity_op` is sometimes a style negative, as in #13813, and this case is *much* more likely to be an outright bug.

### Advantage

Detects code that is unnecessary and may be a bug.

### Drawbacks

As with `identity_op`, the author might want code that completes a regular pattern even if one case is a no-op.

### Example

```rust
pub fn example() -> i32 {
    let mut counter = 0;
    counter += 0;
    counter
}
```

Could be written as:

```rust
pub fn example() -> i32 {
    let mut counter = 0;
    counter
}
```

But the diagnostic should focus on the hypothesis that this is a bug, not that the code can be simplified while preserving its behavior.

**Comment by JaafarTanoukhi** [comment]
I would like to give this a shot. I might work on [#13813](https://github.com/rust-lang/rust-clippy/issues/13813) after, since it's related.
